server/svix-server/src/queue/redis.rs

```diff
@@ -319,6 +342,8 @@ async fn new_pair_inner(
                 tokio::time::sleep(Duration::from_millis(500)).await;
                 continue;
             };
+
+            sleep(Duration::from_millis(500)).await;
```

Comment:
You've now rate limited this worker effectively to once every 0.5s, we should only sleep when nothing is found, not when something is found. The reason for this is because we have a batch size so we may not get everything at once.